tools/flang2/flang2exe/cgmain.cpp

```diff
@@ -1338,7 +1348,36 @@ cons_unroll_count_metadata(int unroll_factor)
   unroll= ll_get_md_node(cpu_llvm_module, LL_PlainMDNode, lvcomp, 2);
   return unroll;
 }
+static LL_MDRef
+cons_vectorlength_enable_metadata( int scalable )
+{
+  LL_MDRef vectorlength = cons_vectorize_metadata();
+  LL_MDRef vectorlength_scalable = cons_vectorlength_scalable_metadata( scalable );
 
+  LL_MDRef md = ll_create_flexible_md_node(cpu_llvm_module);
+  ll_extend_md_node(cpu_llvm_module, md, md);
+  ll_extend_md_node(cpu_llvm_module, md, vectorlength_scalable);
+  ll_extend_md_node(cpu_llvm_module, md, vectorlength);
+  return md;
+}
+static LL_MDRef
+cons_vectorlength_width_metadata( int vectorlength_factor )
+{
+  LL_MDRef lvcomp[2];
+  LL_MDRef width;
+  LL_MDRef vectorize = cons_vectorize_metadata();
+  LL_MDRef vectorlength_scalable = cons_vectorlength_scalable_metadata( 0 );
+  lvcomp[0] = ll_get_md_string( cpu_llvm_module, "llvm.loop.vectorize.width");
+  lvcomp[1] = ll_get_md_i32( cpu_llvm_module, vectorlength_factor );
+  width= ll_get_md_node( cpu_llvm_module, LL_PlainMDNode, lvcomp, 2 );
+  LL_MDRef md = ll_create_flexible_md_node( cpu_llvm_module );
+
+  ll_extend_md_node( cpu_llvm_module, md, md );
```

Comment:
@RichBarton-Arm  I fixed the redundant distinct node in last commit ( https://github.com/flang-compiler/flang/pull/1172/commits/e3b857aaeac278e418956a7a05640a04c0771ac6)
but i still have question what to You mean that we need to support `(<num>,fixed|scalable)`.
I am confused because we now are supporting `vectorlength(fixed)`, `vectorlength(scalable)` and `vectorlength(<num>)`

Reply:
fixed

Reply:
-->fixed